honeybee_radiance_command/_typing.py

```diff
@@ -139,3 +139,72 @@ def normpath(value):
     if ' ' in value:
         value = '{0}{1}{0}'.format(wrapper, value)
     return value
+
+
+def path_checker(file_path, extn_list=None, file_descr=''):
+    """A utility method to check for input file path and normalize the path if present.
+    If extension list and file_descr are provided, then do additional checks to
+    ensure that the correct file type has been specified.
+
+    Args:
+        file_path: The path of the input. Will be normalized through 'typing'
+        extn_list: List of extensions, including the leading dot e.g. ['.hdr','.pic']
+            which if provided will be checked for before the path is set. If there
+            is no match, a value error will be raised.
+        file_descr: A phrase describing the file (e.g. 'Radiance HDR') that can be
+            used to compose the error message in case the extension check fails.
+    """
+
+    if file_path is None:
+        return None
+    else:
+        file_name, file_extn = os.path.splitext(file_path)
+
+        if extn_list:
+            assert type(extn_list) in (list, tuple), \
+                'The input for extn_list(%s) should either be a list or a tuple that ' \
+                'contains extensions (e.g. [".jpg",".bmp"]'
+
+        if extn_list and file_extn.lower() not in extn_list:
+            file_descr = 'for %s files' % file_descr if file_descr else ''
+            err_msg = 'The provided input (%s) is invalid %s as only a file with the ' \
+                      'following extensions is allowed: %s' % (file_path, file_descr,
+                                                               ','.join(extn_list))
+            raise ValueError(err_msg)
+        return normpath(file_path)
```

Comment:
Now I see this and this will actually become a problem. We always use unix-based paths everywhere even the code is executed on Windows. That will avoid issues in cases that a user prepares the recipe on a Windows machine and tries to run in a Unix-based system like Pollination.

This is how it is done in `honeybee-radiance-folder`: 

https://github.com/ladybug-tools/honeybee-radiance-folder/blob/c7dbcc437fdfeb9e25a384c611b307549f4979d9/honeybee_radiance_folder/folderutil.py#L8-L9

Because of this issue I have to request changes for this PR! 😐

Reply:
So shouldnt the changes be applied to the normpath function as well? Because we are applying the windows style path literally everywhere as of now.. [1](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/pcomb.py#L69), [2](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/oconv.py#L69)..

Reply:
Good catch! Yes. We should fix this function: https://github.com/ladybug-tools/honeybee-radiance-command/blob/3c25024836ac9cc69d94ea95955f1aa16fe153d4/honeybee_radiance_command/_typing.py#L136-L141

I'm surprised it hasn't caused issues already. We probably have some additional checks in the executors to fix these cases. I thin we should fix them here too.

Reply:
Okay, so let me fix this first with a separate PR and then I will amend my path tests PR on top of it.